routing/base/astar_algorithm.hpp

```diff
@@ -315,16 +326,24 @@ template <typename TGraph>
 typename AStarAlgorithm<TGraph>::Result AStarAlgorithm<TGraph>::FindPath(
     TGraphType & graph, TVertexType const & startVertex, TVertexType const & finalVertex,
     RoutingResult<TVertexType, TWeightType> & result, my::Cancellable const & cancellable,
-    TOnVisitedVertexCallback onVisitedVertexCallback) const
+    OnVisitedVertexCallback onVisitedVertexCallback, CheckWeightCallback checkWeightCallback) const
 {
   result.Clear();
-  if (nullptr == onVisitedVertexCallback)
+  if (onVisitedVertexCallback == nullptr)
     onVisitedVertexCallback = [](TVertexType const &, TVertexType const &) {};
 
+  if (checkWeightCallback == nullptr)
+    checkWeightCallback = [](TWeightType const & /* weight */) { return true; };
+
   Context context;
   PeriodicPollCancellable periodicCancellable(cancellable);
   Result resultCode = Result::NoPath;
 
+  auto const piS = graph.HeuristicCostEstimate(startVertex, finalVertex);
+  auto const piF = graph.HeuristicCostEstimate(finalVertex, finalVertex);
+  CHECK_LESS(piF, kEpsilon, ("Heuristic is not consistent."));
```

Comment:
```c++
CHECK(my::AlmostEqualAbs(piFi, 0, kEpsilon), ("..."));
```

Reply:
It would be nice to add piF to diagnostics message.

Reply:
done